tools/shared.py

```diff
@@ -265,6 +265,12 @@ def generate_config(path, first_time=False):
 # 4. Fall back users home directory (~/.emscripten).
 
 embedded_config = path_from_root('.emscripten')
+# For compatability with emsdk --embedded mode also look two levels up.  This
+# could be removed in emsdk was use the emscripten directory itself for the
```

Comment:
I don't follow this statement - should `in` be `if`, and `use` be `using`?

But I'm not sure I get it. When are `embedded_config` and `emsdk_embedded_config` different? Clarifying that would be useful here I think.

Reply:
Thanks, fixed the grammar there.

The issue is that emsdk's --embedded mode currently always puts the .emscripton config file in the root of the emsdk, which is two levels up from the emscripten directory.  I'm working on trying to change that but it seems a little complicated.

emscripten already has support for an embedded config, but only its own directory.

This change is only really needs for users who don't run `emsdk_env.sh` and just run `emcc` directly.

Most users are expected to run `emsdk_env.sh` to use the SDK, but it turns out the if you just run `/path/to/emscripten/ecc` without running `emsdk_env.sh` today it often just works because emscripten knowns how to find its config and everything flows from there.   Without this change if you use `emsdk --embedded` and then run `/path/to/emcc` *without* having first run `emsdk_env.sh` then emscripten will not find any config file at all and default back to its method of create new config file with default settings in $HOME.  

This change means that emsdk users won't see this default bahaviour becuse them will look in the correct plance for the embedded config file, even if EM_CONFIG is not set (which it normally is when you run `emsdk_env.sh`.

Sorry for the long explanation, its kind of annoyingly complicated.